modules/RTC/ScreenObtainer.js

```diff
@@ -76,11 +76,9 @@ const ScreenObtainer = {
         this.options = options = options || {};
         gumFunction = gum;
 
-        if (RTCBrowserType.isFirefox()) {
-            initFirefoxExtensionDetection(options);
-        }
-
-        if (RTCBrowserType.isNWJS()) {
+        if (this.options.disableDesktopSharing) {
```

Comment:
I had the same thoughts as you when trying to add this initially, but it's not as simple as it seems.

If I add this at the very beginning with some return statement then I will print the message about desktop sharing disabled twice, because it can be disabled during this 'if else if' chain execution (disabled on one browser).

If I add this at the end then all extension detection will be executed for nothing (possibly registering some handlers - I don't know if it's done now, but can be added later and we'll have bugs).

So I found it's the easies to abort this long else if chain, but I can wrap that long "if else if" into separate method and maybe something better will come out of that.